Mathlib/Geometry/Manifold/IntegralCurve.lean

```diff
@@ -0,0 +1,551 @@
+    (hγ : IsIntegralCurveAt γ v t₀) (hγ' : IsIntegralCurveAt γ' v t₀) (h : γ t₀ = γ' t₀) :
+    ∃ ε > 0, EqOn γ γ' (Ioo (t₀ - ε) (t₀ + ε)) := by
+  -- first define `v'` as the vector field expressed in the local chart around `γ t₀`
+  -- this is basically what the function looks like when `hv` is unfolded
+  set v' : E → E := fun x =>
+    tangentCoordChange I ((extChartAt I (γ t₀)).symm x) (γ t₀) ((extChartAt I (γ t₀)).symm x)
+      (v ((extChartAt I (γ t₀)).symm x)) with hv'
+
+  -- extract set `s` on which `v'` is Lipschitz
+  rw [contMDiffAt_iff] at hv
+  obtain ⟨_, hv⟩ := hv
+  obtain ⟨K, s, hs, hlip⟩ : ∃ K, ∃ s ∈ nhds _, LipschitzOnWith K v' s :=
+    ContDiffAt.exists_lipschitzOnWith (hv.contDiffAt (range_mem_nhds_isInteriorPoint I hγt₀)).snd
+  have hlip : ∀ t : ℝ, LipschitzOnWith K ((fun _ => v') t) ((fun _ => s) t) := fun _ => hlip
+
+  -- `γ t` when expressed in the local chart should remain inside `s`
+  have hcont : ContinuousAt ((extChartAt I (γ t₀)) ∘ γ) t₀ :=
+    (continuousAt_extChartAt ..).comp hγ.continuousAt
+  rw [continuousAt_def] at hcont
+  have hnhds := hcont _ hs
+  rw [← eventually_mem_nhds] at hnhds
+
+  -- `γ t` should remain inside the domain of the local chart around `γ t₀`
+  have hsrc := continuousAt_def.mp hγ.continuousAt _ <| extChartAt_source_mem_nhds I (γ t₀)
+  rw [← eventually_mem_nhds] at hsrc
+
+  -- `γ` is tangent to `v` in some neighbourhood of `t₀`
+  simp_rw [IsIntegralCurveAt, IsIntegralCurveOn, ← Filter.eventually_iff_exists_mem] at hγ
+
+  -- same as above but for `γ'`
+  have hcont' : ContinuousAt ((extChartAt I (γ' t₀)) ∘ γ') t₀ :=
+    ContinuousAt.comp (continuousAt_extChartAt ..) hγ'.continuousAt
+  rw [continuousAt_def] at hcont'
+  have hnhds' := hcont' _ (h ▸ hs)
+  rw [← eventually_mem_nhds] at hnhds'
+
+  have hsrc' := continuousAt_def.mp hγ'.continuousAt _ <| extChartAt_source_mem_nhds I (γ' t₀)
+  rw [← eventually_mem_nhds] at hsrc'
+
+  simp_rw [IsIntegralCurveAt, IsIntegralCurveOn, ← Filter.eventually_iff_exists_mem] at hγ'
+
+  -- there exists a neighbourhood around `t₀` in which all of the above hold
+  have haux := hnhds.and <| hsrc.and <| hγ.and <| hnhds'.and <| hsrc'.and hγ'
+  rw [Metric.eventually_nhds_iff_ball] at haux
+
+  obtain ⟨ε, hε, haux⟩ := haux
+  refine ⟨ε, hε, ?_⟩
+
+  -- break out all the conditions again
+  have hmem := fun t ht => mem_preimage.mp <| mem_of_mem_nhds (haux t ht).1
+  have hsrc := fun t ht => mem_preimage.mp <| mem_of_mem_nhds (haux t ht).2.1
+  have hmfd : IsIntegralCurveOn _ _ _ := fun t ht => (haux t ht).2.2.1
+  have hmem' := fun t ht => mem_preimage.mp <| mem_of_mem_nhds (haux t ht).2.2.2.1
+  have hsrc' := fun t ht => mem_preimage.mp <| mem_of_mem_nhds (haux t ht).2.2.2.2.1
+  have hmfd' : IsIntegralCurveOn _ _ _ := fun t ht => (haux t ht).2.2.2.2.2
+
+  -- `γ` and `γ'` when expressed in the local chart are continuous on this neighbourhood
+  have hcont := (continuousOn_extChartAt I (γ t₀)).comp
```

Comment:
You can use dot notation again.
`have hcont := (continuousOn_extChartAt I (γ t₀)).comp hmfd.continuousOn hsrc`